### PR TITLE
Support all signing algorithms for backwards compatibility

### PR DESCRIPTION
**Pull request by enj**
## Description

This change restores the implicit behavior of [github.com/coreos/go-oidc/v3/oidc.Provider.Verifier(...)](https://github.com/coreos/go-oidc/blob/6ce86d9e30589a68cbf891b7c4fc0d316201b786/oidc/verify.go#L110-L116) which automatically trusts all algorithms supported by the issuer.

Also drops use of github.com/dgrijalva/jwt-go

Both of these regressions were caused by #1559

Signed-off-by: Monis Khan <mok@vmware.com>

## Motivation and Context

Fix regressions (that have not yet been released).

## How Has This Been Tested?

Ran unit tests and manually tested it with an issuer that uses `ES256` instead of the standard `RS256`.

## Checklist:

- [ ] My change requires a change to the documentation or CHANGELOG.
- [ ] I have updated the documentation/CHANGELOG accordingly.
- [ ] I have created a feature (non-master) branch for my PR.

cc @JoelSpeed

### CODE REVIEW

File: pkg/providers/oidc/provider_verifier.go [hunk 1]
```diff
@@ -79,6 +79,19 @@ func (p ProviderVerifierOptions) toOIDCConfig() *oidc.Config {
 		ClientID:          p.ClientID,
 		SkipIssuerCheck:   p.SkipIssuerVerification,
 		SkipClientIDCheck: true,
+		// support all signing algorithms for backwards compatibility
+		// there is no security value in this config, either you trust the issuer or you do not
```
Comment (JoelSpeed) [inline pkg/providers/oidc/provider_verifier.go:83]: Why do we not want to limit this to explicitly those supported by the provider? Surely given we have this information from the discovery document we should pass it through here?

**Review by JoelSpeed — COMMENTED** [review]
To properly restore the behaviour, shouldn't we be passing through the supported algorithms from the discovery document rather than just saying every known type is supported?

What are the implications here of saying every known type is supported? Does it impact the performance?

**Comment by JoelSpeed** [comment]
@enj Do you have any response to my previous review? Keen to get this fix in if we can

**Comment by enj** [comment]
> @enj Do you have any response to my previous review? Keen to get this fix in if we can

@JoelSpeed I probably won't have time to get back to this anytime soon.  I am going to close this - anyone is free to pick it up if they want to.

**Pull request closed without merging**